### PR TITLE
Document proposed shipping methods v3 API

### PR DESCRIPTION
**Pull request by bc-luke**
Draft shipping methods API to support new shipping promotions. This doesn't need to be public but perhaps we will choose to release it publicly. To be discussed.

image

## What changed?

Document the parts of a shipping methods API that would be necessary to support the new shipping promotions UI including: 
* Filter by name, channel ID and zone
* Optionally include zone information (zone name and ID) and channel IDs
* Method name
* Method ID

## Release notes draft

* 

## Anything else?

ping {names}

### CODE REVIEW

File: reference/shipping.v3.yml [hunk 1]
```diff
@@ -355,6 +440,41 @@ components:
...
+          description: The unique identifier of the shipping method.
+          type: integer
+          format: int32
+          example: 2
+        name:
+          description: The human-readable name of the shipping method.
+          type: string
+          example: "Courier delivery"
+        shipping_zone:
+          description: Zone information. This is available only when "include=shipping_zone" is present in the querystring.
```
Comment (theromulans) [inline reference/shipping.v3.yml:459]: Seems like a sensible optional include 👌

File: reference/shipping.v3.yml [hunk 2]
```diff
@@ -355,6 +440,41 @@ components:
         - commodity_description
         - international_shipping
         - hs_codes
+    shippingMethods:
+      title: shippingMethods
+      description: Data about the customs information object.
```
Comment (theromulans) [inline reference/shipping.v3.yml:445]: Oops!

File: reference/shipping.v3.yml [hunk 3]
```diff
@@ -355,6 +440,41 @@ components:
...
+      type: object
+      x-internal: false
```
Comment (theromulans) [inline reference/shipping.v3.yml:447]: What does this represent?

File: reference/shipping.v3.yml [hunk 4]
```diff
@@ -355,6 +440,41 @@ components:
...
+              description: The unique identifier of the shipping zone.
+              type: integer
+              format: int32
+              example: 2
+            name:
+              description: The human-readable name of the shipping zone.
+              type: string
+              example: "Thailand"
+        channel_ids:
+          description: Channel information. This is available only when "include=shipping_zone" is present in the querystring.
```
Comment (theromulans) [inline reference/shipping.v3.yml:472]: Would we not want to always include the Channel IDs? Well, perhaps only when the store has multiple channels?

File: reference/shipping.v3.yml (hunk 3 above)
Comment (vborinsk) [reply reference/shipping.v3.yml:447]: Let's make it public, I have seen at least a few requests from agencies to have an API to query all methods without having to iterate over all the zones.

File: reference/shipping.v3.yml [hunk 5]
```diff
@@ -20,6 +20,91 @@ servers:
...
+          name: cursor
+          description: 'Encoded cursor'
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: include
+          description: |-
+            * `shipping_zone` - shipping zone information
+            * `shipping_methods.channel_ids` - IDs of the associated channels
```
Comment (agchan12) [inline reference/shipping.v3.yml:98]:
```suggestion
          description: |-
            * `shipping_zone` - Include shipping zone information with each shipping method.
            * `shipping_methods.channel_ids` - Include the IDs of the associated channels for each shipping method.

```

File: reference/shipping.v3.yml (hunk 4 above)
Comment (bc-luke) [reply reference/shipping.v3.yml]: Yes probably a reasonable expecation. Updated!

File: reference/shipping.v3.yml (hunk 5 above)
Comment (bc-luke) [reply reference/shipping.v3.yml]: 🔧Fixed

File: reference/shipping.v3.yml (hunk 2 above)
Comment (bc-luke) [reply reference/shipping.v3.yml]: Fixed

File: reference/shipping.v3.yml (hunk 3 above)
Comment (bc-luke) [reply reference/shipping.v3.yml]: In that case, it might make sense to add more properties so we have feature parity with V2.

**Pull request closed without merging**